app/main.py

```diff
@@ -1,6 +1,20 @@
-from typing import Callable
+from typing import Callable, Dict, Tuple
+import functools
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    func_cache: Dict[Tuple, any] = {}
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs) -> Callable:
+        key = (args, frozenset(kwargs.items()))
+        if key in func_cache:
+            print("Getting from cache")
+            return func_cache[key]
+        else:
+            print("Calculating new result")
+            result = func(*args, **kwargs)
+            func_cache[key] = result
+            return result
```

Comment:
```suggestion
        if key in func_cache:
            print("Getting from cache")
        else:
            print("Calculating new result")
            func_cache[key] = func(*args, **kwargs)
        return func_cache[key]
```
Your code can be simplified a bit))